.golangci.yml

```diff
@@ -90,90 +84,119 @@ linters:
-        - wrapcheck
-    - text: "error returned from external package is unwrapped: sig: func github.com/kopia/kopia/internal/connection.UsingConnection"
-      linters:
-        - wrapcheck
-    - text: timeCmpSimplify
-      linters:
-        - gocritic
-    - text: ".*Magic number\\: [01],"
-      linters:
-        - mnd
-    - text: "Errors unhandled"
-      linters:
-        - gosec
-    - text: "unwrapped: sig: func github.com/kopia/kopia/fs.GetAllEntries"
-      linters:
-        - wrapcheck
-    - text: "float-compare: use require.InEpsilon"
-      linters:
-        - testifylint
+  exclusions:
+    generated: lax
+    rules:
+      - path: reporter.go
+        linters:
+          - musttag        
+      - path: _test\.go|testing|tests|test_env|fshasher|fault 
+        linters:
+          - contextcheck
+          - errcheck
+          - errchkjson
+          - forcetypeassert
+          - funlen
+          - gochecknoglobals
+          - gocognit
+          - goconst
+          - gosec
+          - mnd
+          - musttag
+          - nestif
+          - nolintlint
+          - perfsprint
+          - revive
+          - wrapcheck
+          - wsl
+      - text: "log is a global variable"
+        linters:
+          - gochecknoglobals
+      - text: "tracer is a global variable"
+        linters:
+          - gochecknoglobals
+        # always allow ctx even when unused
+      - text: "unused-parameter: parameter 'ctx' seems to be unused"
+        linters:
+          - revive
+      - text: "Magic number: 1e"
+        linters:
+          - mnd
+      - text: "unnecessaryDefer"
+        linters:
+          - gocritic
+      - text: "filepathJoin"
+        linters:
+          - gocritic
+      - text: "weak cryptographic primitive"
+        linters:
+          - gosec
+      - text: "lines are duplicate of"
+        linters:
+          - dupl
+      - text: "Line contains TODO"
+        linters:
+          - godox
+      - text: "error returned from external package is unwrapped: sig: func github.com/kopia/kopia/internal/retry"
+        linters:
+          - wrapcheck
+      - text: "error returned from external package is unwrapped: sig: func github.com/kopia/kopia/internal/connection.UsingConnection"
+        linters:
+          - wrapcheck
+      - text: timeCmpSimplify
+        linters:
+          - gocritic
+      - text: ".*Magic number\\: [01],"
+        linters:
+          - mnd
+      - text: "Errors unhandled"
+        linters:
+          - gosec
+      - text: "unwrapped: sig: func github.com/kopia/kopia/fs.GetAllEntries"
+        linters:
+          - wrapcheck
+      - text: "float-compare: use require.InEpsilon"
+        linters:
+          - testifylint
+      - text: "fieldalignment:"
+        linters:
+          - govet
+      - text: "shadow:"
+        linters:
+          - govet
+      - text: "unnamedResult:"
+        linters:
+          - gocritic
+      - linters:
+          - cyclop
+        path: (.+)_test\.go
```

Comment:
this could probably be included in the list of linter for the rule with:
```yaml
- path: _test\.go|testing|tests|test_env|fshasher|fault
```

Reply:
Here's the current rationale, but it can definitely be revisited and refined.
- The paths mentioned in that rule are all packages (or files) only used in testing.
- `cyclop` (and `gocyclo`) end up being being noisy with test functions.

Now, given that this is only being applied to the narrower scope already and is not flagging anything, it makes sense to leave it here.

Thanks for the feedback 😉